### ISSUE TITLE
Differentiating between object reference and class reference

### ISSUE THREAD

**Issue by Kein** [guttir14/UnrealDumper-4.25#72]
So I have this dump from UD:

```c++
struct AMyGameMode : AGameModeBase {
	struct TArray<struct AMyCharacterBase*> ListOfPlayables; // 0x320(0x10)
```

However, in actuality it is
`TArray<TSubclassOf<AMyCharacterBase>>`

These are completely different things. Is it impossible to differentiate this type runtime when dumping?

**Comment by guttir14** [comment]
So, in example you provided there is array of UClass objects? Could you give me game name, class and field name where this bug occurs, so I can find a way to differentiate them.

**Comment by Kein** [comment]
> So, in example you provided there is array of UClass pointers?

No, it is a type. For example, In UE editor in Blueprints, when you create a variable to an reference type UObject like Actor or PlayerController it offers 2 choices:

![image](https://user-images.githubusercontent.com/623756/153130538-5da84ff4-e15a-4285-afda-5dd2ce550637.png)

`Object Reference` and `Class Reference` (soft-references just some weak-pointer wrappers)

On C++ side it is respectively:
`TArray<AMyCharacterBase*>` for an array of references/pointer of the objects of said class
and
`TArray<TSubclassOf<AMyCharacterBase>>` is an array of class types

> Could you give me game name, class and field name where this bug occurs, so I can find a way to differentiate them.

I can but I believe you dont need to know specific game, it is nothing unique and this is just how UDumper seems to handle things in general (?). 

If you desire so I can make repro test project on UE 4.26.2, it is very simple, just define basic class with 2 types of said members and then package and dump. The only way I managed to figure this out is because they have used it in a blueprint that inherited custom C++ class and in BP they have populated inherited field which was serialized later with BP body. Obviously you cant do that with pointers.

**Comment by Kein** [comment]
Test project:

https://www.dropbox.com/s/u8soy5cjtu9cph4/WindowsNoEditor.7z?dl=1

```c++
// Fill out your copyright notice in the Description page of Project Settings.

#pragma once

#include "CoreMinimal.h"
#include "GameFramework/Actor.h"
#include "MyActor.generated.h"

UCLASS()
class MYPROJECT_API AMyActor : public AActor
{
	GENERATED_BODY()
	
public:	
	// Sets default values for this actor's properties
	AMyActor();

protected:
	// Called when the game starts or when spawned
	virtual void BeginPlay() override;

public:	
	// Called every frame
	virtual void Tick(float DeltaTime) override;

	UPROPERTY(EditAnywhere, BlueprintReadWrite)
	TArray<AMyActor*> ObjectReference;

	UPROPERTY(EditAnywhere, BlueprintReadWrite)
	TArray<TSubclassOf<AMyActor>> TypeReference;

};
```

Dumped as:
```c++
// Class MyProject.MyActor
// Size: 0x240 (Inherited: 0x220)
struct AMyActor : AActor {
	struct TArray<struct AMyActor*> ObjectReference; // 0x220(0x10)
	struct TArray<struct AMyActor*> TypeReference; // 0x230(0x10)
};
```

**Comment by WhiteHair-Chen** [comment]
So, is this problem solved?
